### PR TITLE
refactored function node desc to have cleaner input and less code

### PR DESCRIPTION
**Pull request by oveddan**
Did some refactoring on `FunctionDesc` to make it more readable and simpler to consume:

* Got rid of being able to have inputs/outputs be declared either as an array or key<->value, its confusing to have an api that can be called in multiple ways like this.  I know the desire is to move to key<-> value pairs, but it doesn't seem like that is being used now for function node, so why don't we wait til we do it the right  way (i.e. as top level api)  first.
* Refactored the code to have a reusable function: `toOrderedSockets` that converts from Array<valueTypeName> -> Socket, with an argument for how to determine the keys of each element.

### CODE REVIEW

File: packages/core/src/Nodes/FunctionNode.ts [hunk 1]
```diff
@@ -28,71 +28,76 @@ export class FunctionNode extends Node {
...
+    name,
+    label,
+    aliases,
+    exec
+  }: {
     name: string;
     label?: string;
     category?: NodeCategory;
     aliases?: string[];
-    in?: { [name: string]: string }[] | string[];
```
Comment (oveddan) [inline packages/core/src/Nodes/FunctionNode.ts:37]: Until we are calling it with key value pair, lets not expose this as an argument;

**Comment by oveddan** [comment]
Ok looking thru the code it looks like clamp passes an array of objects...let me revisit.  Do not merge

**Comment by bhouston** [comment]
Love the simplification!  Thank you!

**Comment by bhouston** [comment]
BTW This just got merged with the other PR.  I think it works though.

**Pull request merged**